api/src/test/java/org/openmrs/api/impl/ConceptServiceImplTest.java

```diff
@@ -288,6 +288,44 @@ public void saveConcept_shouldForceSetFlagIfSetMembersExist() {
 		assertTrue(concept.getSet());
 	}
 	
+	/**
+	 * @see ConceptServiceImpl#saveDrug(Drug) 
+	 */
+	@Test
+	public void saveDrug_shouldPutGeneratedIdOntoReturnedDrug() {
+		executeDataSet("org/openmrs/api/include/ConceptServiceTest-initialConcepts.xml");
+		Drug drug = new Drug();
+		drug.setConcept(conceptService.getConcept(1));
+		Drug savedDrug = conceptService.saveDrug(drug);
+		assertNotNull(savedDrug.getDrugId());
+	}
+
+	/**
+	 * @see ConceptServiceImpl#saveDrug(Drug)
+	 */
+	@Test
+	public void saveDrug_shouldCreateNewDrugInDatabase() {
+		executeDataSet("org/openmrs/api/include/ConceptServiceTest-initialConcepts.xml");
+		Drug drug = new Drug();
+		drug.setConcept(conceptService.getConcept(1));
+		Drug savedDrug = conceptService.saveDrug(drug);
+		assertNotNull(conceptService.getDrug(savedDrug.getDrugId()));
+	}
+
+	/**
+	 * @see ConceptServiceImpl#saveDrug(Drug)
+	 */
+	@Test
+	public void saveDrug_shouldUpdateDrugAlreadyExistingInDatabase() {
+		executeDataSet("org/openmrs/api/include/ConceptServiceTest-initialConcepts.xml");
+		Drug drug = new Drug();
+		drug.setConcept(conceptService.getConcept(1));
+		Drug savedDrug = conceptService.saveDrug(drug);
+		savedDrug.setCombination(true);
```

Comment:
Before this line, You should check that this Drug object `getCombination` is false. Otherwise, How can you ensure that it was updated by this line?